src/main/scala/it/pagopa/interop/notifier/service/converters/AgreementEventsConverter.scala

```diff
@@ -13,26 +13,30 @@ object AgreementEventsConverter {
   def getMessageId(dynamoService: DynamoNotificationResourcesService)(implicit
     ec: ExecutionContext,
     contexts: Seq[(String, String)]
-  ): PartialFunction[ProjectableEvent, Future[Option[MessageId]]] = { case e: Event =>
+  ): PartialFunction[ProjectableEvent, Future[Seq[MessageId]]] = { case e: Event =>
     getMessageIdFromEvent(dynamoService, e)
   }
 
   private[this] def getMessageIdFromEvent(dynamoService: DynamoNotificationResourcesService, event: Event)(implicit
     ec: ExecutionContext,
     contexts: Seq[(String, String)]
-  ): Future[Option[MessageId]] = event match {
-    case AgreementAdded(a)                       =>
-      val messageId: MessageId = MessageId(a.id, a.producerId.toString())
-      dynamoService.put(messageId).map(_ => messageId.some)
-    case AgreementDeleted(id)                    => getMessageIdFromDynamo(dynamoService)(id).map(_.some)
-    case AgreementUpdated(a)                     => Future.successful(MessageId(a.id, a.producerId.toString()).some)
-    case AgreementConsumerDocumentAdded(id, _)   => getMessageIdFromDynamo(dynamoService)(id).map(_.some)
-    case AgreementConsumerDocumentRemoved(id, _) => getMessageIdFromDynamo(dynamoService)(id).map(_.some)
-    case VerifiedAttributeUpdated(a)             => Future.successful(MessageId(a.id, a.producerId.toString()).some)
-    case AgreementActivated(a)                   => Future.successful(MessageId(a.id, a.producerId.toString()).some)
-    case AgreementSuspended(a)                   => Future.successful(MessageId(a.id, a.producerId.toString()).some)
-    case AgreementDeactivated(a)                 => Future.successful(MessageId(a.id, a.producerId.toString()).some)
-    case AgreementContractAdded(id, _)           => getMessageIdFromDynamo(dynamoService)(id).map(_.some)
+  ): Future[Seq[MessageId]] = event match {
+    case AgreementAdded(a)                       => 
+      val messageId = MessageId(a.id, a.producerId.toString())
+      val messageAgreements = MessageId(a.id, agreements)
+      for {
+      _ <- dynamoService.put(messageId)
+      _ <-  dynamoService.put(messageAgreements)
```

Comment:
unfortunately I think this would not work.
`a.id` is used as key in dynamo table, so the second `put` overrides the first.

I have a solution in mind but let me sleep on it and see if it still makes sense tomorrow

Reply:
We can use the second parameter as a sort key.